src/components/figure.js

```diff
@@ -1,43 +1,40 @@
 /**
  * Figure
  * An individual gallery tile
+ * @flow
  */
 
 import React from 'react'
 import Image from './ui/image'
 import Ink from 'react-ink'
 import cx from 'classnames'
-import createClass from 'create-react-class'
-import { object, func, bool } from 'prop-types'
+import { type Record } from '../record'
 
-let Figure = createClass({
-  propTypes: {
-    record: object.isRequired,
-    onClick: func.isRequired,
-    picked: bool
-  },
+type Props = {
+  record: Record,
+  onClick: string => *,
+  picked: boolean
+}
 
+export default class Figure extends React.Component<Props> {
   render() {
     let { record, picked } = this.props
 
-    let className = cx({
-      'ars-fig': true,
+    let className = cx('ars-fig', {
       'ars-fig-picked': picked
     })
 
     return (
-      <button className={className} onClick={this._onClick}>
+      <button className={className} onClick={this._onClick.bind(this)}>
         <Image className="ars-fig-img" src={record.url} />
         <span className="ars-fig-caption">{record.name}</span>
         <Ink opacity={0.4} />
       </button>
     )
-  },
+  }
 
-  _onClick(e) {
-    e.preventDefault()
+  _onClick(event: MouseEvent) {
```

Comment:
any chance this could also be a keyboard event?

Reply:
Good call. Also this should use the synthetic event system types

Reply:
I'll do the synthetic typing upstream. For now, I'll generalize this to `Event` since it only needs that interface.